src/Main.elm

```diff
@@ -42,27 +68,50 @@ update msg model =
 
 view : Model -> Html Msg
 view model =
-    div [ class "board" ] (viewBoardSquares model.board)
+    div [ class "board" ] (viewBoardSquares model.board model.selectedSquare)
 
 
-viewBoardSquares : List Int -> List (Html Msg)
-viewBoardSquares board =
+viewBoardSquares : List BoardSquare -> Maybe Int -> List (Html Msg)
+viewBoardSquares board selectedSquare =
     let
+        cols : Int
         cols =
             round (sqrt (toFloat (List.length board)))
 
+        rows : Int
         rows =
             round (sqrt (toFloat (List.length board)))
     in
     board
         |> List.map
-            (\squareNumber ->
-                div
-                    [ class "flex flex-col justify-center h-24 border border-black border-solid text-center hover:bg-gray-400"
-                    , onClick (SquareClicked squareNumber)
-                    ]
-                    [ text (boardNumberToSquareNumber cols squareNumber ++ boardNumberToSquareLetter cols squareNumber)
-                    ]
+            (\square ->
+                let
+                    baseClassList =
+                        "cursor-pointer flex flex-col justify-center h-24 border border-black border-solid text-center hover:bg-gray-400"
+
+                    isSelected : Bool
+                    isSelected =
+                        selectedSquare
+                            |> Maybe.map (\selectedSquareNumber -> selectedSquareNumber == square.id)
+                            |> Maybe.withDefault False
+                in
+                if isSelected then
```

Comment:
If the square was selected, show a replacement tile with two buttons that will lay the tile or cancel. In reality these interactions would be moved to a different part of the UI, but I didn't want to build out more UI and mess with grid layout right now